testdata/nil-len.go

```diff
@@ -35,4 +35,6 @@ func fn() {
 
 	if s != nil && len(s) == 0 { // nil check is not redundant here
 	}
+	if s != nil && len(s) != 1 { // nil check is not redundant here
```

Comment:
Is it, though? the truth tables for `s != nil && len(s) != 1` and `len(s) != 1` are different. You can either have a non-nil slice of length 0, or a nil slice of length 0.